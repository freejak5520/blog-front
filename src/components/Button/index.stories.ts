import type { Meta, StoryObj } from "@storybook/react";
import { fn } from "@storybook/test";
import Button from ".";

const meta = {
  title: "Components/Common/Button",
  component: Button,
  parameters: {
    layout: "centered",
  },

  tags: ["autodocs"],

  argTypes: {
    children: { type: "string", description: "text" },
    onClick: {
      type: "function",
      description: "On click action",
    },
  },

  args: {
    children: "Button",
    onClick: fn(),
  },
} satisfies Meta<typeof Button>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Primary: Story = {};
