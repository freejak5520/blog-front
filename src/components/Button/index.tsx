"use client";

import { ButtonHTMLAttributes } from "react";
import { useFormStatus } from "react-dom";

const Button = ({
  type,
  children,
}: {
  type?: ButtonHTMLAttributes<HTMLButtonElement>["type"];
  children?: React.ReactNode;
}) => {
  const { pending } = useFormStatus();

  return (
    <button
      type={type}
      className="border-r bg-blue-800 text-white font-bold px-4 py-3.5 leading-5 text-sm rounded w-full disabled:bg-gray-400"
      disabled={pending}
    >
      {children}
    </button>
  );
};

export default Button;
