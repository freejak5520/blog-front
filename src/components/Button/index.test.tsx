import { render } from "@testing-library/react";
import Button from "@/components/Button";
import "@testing-library/jest-dom";

jest.mock("react-dom", () => ({
  ...jest.requireActual("react-dom"),
  useFormState: () => [{}, () => {}],
  useFormStatus: () => ({ pending: false }),
}));

describe("Button", () => {
  it("render button", () => {
    const { container } = render(<Button>hello</Button>);

    expect(container).toMatchSnapshot();
  });
});
