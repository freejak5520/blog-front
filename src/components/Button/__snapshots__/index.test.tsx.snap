// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button render button 1`] = `
<div>
  <button
    class="w-full rounded border-r bg-blue-800 px-4 py-3.5 text-sm font-bold leading-5 text-white disabled:bg-gray-400 undefined"
  >
    hello
  </button>
</div>
`;
